### DISCUSSION TITLE
Feature idea

### DISCUSSION THREAD

**Discussion by MichelNivard** [larsbaunwall/bricky#3] [Ideas]
HI Lars,

among the sprawl of "chat + doc search" prototypes you build on of the best! I really like that it pulls .md from a github (or anywhere I supose) because its very flexible, I am a scientist and there are 20 options for me to convert a folder of PDF to .md and point bricky at that folder fort a constrained search.

2 feature requests:
a. have bricky print the source docs context it got the answer from and the links to the files? handy if ppl want to do more reading on the topic
b.  bit of luxury but let bricky index with any hugginface/transformer model you could run locally/on your server, because thats going to become the expensive step if you have ±5k docs?

amazing UI, love it!

Best,
Michel

**Comment by larsbaunwall** [comment]
Hi @MichelNivard,

Thanks for taking the time to post. Great ideas! I actually have the same on my to-do regarding the sources. Essentially mimicking the awesome [Housten AI](http://houston.astro.build) built by Astro on a similar architecture. It requires a slightly expanded MarkdownConverter that picks up the file path and puts it into the Document's metadata.

Regarding the pipelines - you can easily change them for your own custom ones - or [standard haystack nodes](https://docs.haystack.deepset.ai/docs/nodes_overview) that wrap models from HuggingFace. You can e.g. tweak the [EmbeddingRetriever](https://github.com/larsbaunwall/bricky/blob/7a398bd1b7cdae361554bcc51fad7d38aa633f4c/api/pipelines/indexing.py#L62) to use a HuggingFace model instead (see [this](https://docs.haystack.deepset.ai/docs/retriever#embedding-retrieval-recommended))

See [the pipelines folder](https://github.com/larsbaunwall/bricky/tree/main/api/pipelines) for to explore the two pipelines used in Bricky currently.

**Reply by larsbaunwall** [reply]
Source doc snippets and file paths just made it through to the API in https://github.com/larsbaunwall/bricky/commit/17b9130429596468de15e5278f51fa7a0f8a2c23. Now the UI just need a bit of love 😄
